tests/particles/insert_particle.cc

```diff
@@ -0,0 +1,82 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+// check that inserting particles works also in parallel when one processor is
+// not inserting any particles
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/fe/mapping_q.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+
+#include <deal.II/particles/generators.h>
+#include <deal.II/particles/particle_handler.h>
+
+#include "../tests.h"
+
+template <int dim, int spacedim>
+void
+test()
+{
+  {
+    parallel::distributed::Triangulation<dim, spacedim> tr(MPI_COMM_WORLD);
+
+    GridGenerator::hyper_cube(tr);
+    tr.refine_global(1);
+    MappingQ<dim, spacedim> mapping(1);
+
+    Particles::ParticleHandler<dim, spacedim> particle_handler(tr, mapping);
+
+    std::vector<Point<dim>> positions(1, Point<dim>());
+    if (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 1)
+      positions.clear();
+
+    particle_handler.insert_particles(positions);
+
+    deallog << "Particle number (global): "
+            << particle_handler.n_global_particles() << std::endl;
+    deallog << "Particle number (local): "
+            << particle_handler.n_locally_owned_particles() << std::endl;
+
+    for (const auto &particle : particle_handler)
+      {
+        deallog << "Particle id: " << particle.get_id() << std::endl;
+      }
+  }
+
+  deallog << "OK" << std::endl;
+}
+
+
+
+int
+main(int argc, char *argv[])
+{
+  Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, 1);
+
+  MPILogInitAll init;
```

Comment:
The test looks great, including a short description of what the test covers. One thing one could think about is whether we need this init-all structure or whether a simple `mpi_initlog()` would suffice. Then we only get information from rank 0, but I think it would be good enough as we do not really need much output for this test, and I find it better to have as little output as possible to see whether we have pass/fail, and scrolling up/down is a bit of extra work.

Reply:
Yes, that sounds reasonable. I update that soon.